components/canvas/canvas_data.rs

```diff
@@ -0,0 +1,969 @@
+        let anx = (cp1.x - cp0.x) / a2.sqrt();
+        let any = (cp1.y - cp0.y) / a2.sqrt();
+        let tp1 = Point2D::new(cp1.x - anx * d, cp1.y - any * d);
+
+        // second tangent point
+        let bnx = (cp1.x - cp2.x) / b2.sqrt();
+        let bny = (cp1.y - cp2.y) / b2.sqrt();
+        let tp2 = Point2D::new(cp1.x - bnx * d, cp1.y - bny * d);
+
+        // arc center and angles
+        let anticlockwise = direction < 0.0;
+        let cx = tp1.x + any * radius * if anticlockwise { 1.0 } else { -1.0 };
+        let cy = tp1.y - anx * radius * if anticlockwise { 1.0 } else { -1.0 };
+        let angle_start = (tp1.y - cy).atan2(tp1.x - cx);
+        let angle_end = (tp2.y - cy).atan2(tp2.x - cx);
+
+        self.line_to(&tp1);
+        if [cx, cy, angle_start, angle_end].iter().all(|x| x.is_finite()) {
+            self.arc(&Point2D::new(cx, cy), radius,
+                     angle_start, angle_end, anticlockwise);
+        }
+    }
+
+    pub fn ellipse(&mut self,
+           center: &Point2D<AzFloat>,
+           radius_x: AzFloat,
+           radius_y: AzFloat,
+           rotation_angle: AzFloat,
+           start_angle: AzFloat,
+           end_angle: AzFloat,
+           ccw: bool) {
+        self.path_builder.ellipse(*center, radius_x, radius_y, rotation_angle, start_angle, end_angle, ccw);
+    }
+
+    pub fn set_fill_style(&mut self, style: FillOrStrokeStyle) {
+        if let Some(pattern) = style.to_azure_pattern(&self.drawtarget) {
+            self.state.fill_style = pattern
+        }
+    }
+
+    pub fn set_stroke_style(&mut self, style: FillOrStrokeStyle) {
+        if let Some(pattern) = style.to_azure_pattern(&self.drawtarget) {
+            self.state.stroke_style = pattern
+        }
+    }
+
+    pub fn set_line_width(&mut self, width: f32) {
+        self.state.stroke_opts.line_width = width;
+    }
+
+    pub fn set_line_cap(&mut self, cap: LineCapStyle) {
+        self.state.stroke_opts.line_cap = cap.to_azure_style();
+    }
+
+    pub fn set_line_join(&mut self, join: LineJoinStyle) {
+        self.state.stroke_opts.line_join = join.to_azure_style();
+    }
+
+    pub fn set_miter_limit(&mut self, limit: f32) {
+        self.state.stroke_opts.miter_limit = limit;
+    }
+
+    pub fn set_transform(&mut self, transform: &Transform2D<f32>) {
+        self.state.transform = transform.clone();
+        self.drawtarget.set_transform(transform)
+    }
+
+    pub fn set_global_alpha(&mut self, alpha: f32) {
+        self.state.draw_options.alpha = alpha;
+    }
+
+    pub fn set_global_composition(&mut self, op: CompositionOrBlending) {
+        self.state.draw_options.set_composition_op(op.to_azure_style());
+    }
+
+    pub fn create(size: Size2D<i32>) -> DrawTarget {
+        DrawTarget::new(BackendType::Skia, size, SurfaceFormat::B8G8R8A8)
+    }
+
+    pub fn recreate(&mut self, size: Size2D<i32>) {
+        // TODO: clear the thread state. https://github.com/servo/servo/issues/17533
```

Comment:
This probably needs a different comment now?

Reply:
Oh. I didn't even really look at the code I was moving. Looking at that issue, it seems like it was fixed last year and the comment snuck its way back in. Should I just remove it?

Reply:
Ended up deciding to just remove it. I looked back at the issue and it has been closed since last July (also the code immediately below it does what it says it needs to do). Pretty sure that comment just snuck back in at some point. Just pushed that alongside the hashmap changes.